lib/fastlane/plugin/wpmreleasetoolkit/actions/common/openai_generate_action.rb

```diff
@@ -0,0 +1,141 @@
+          Do not write it point by point and keep it under 350 characters. It should be a unique paragraph.
+
+          When provided a list, use the number of any potential "*" in brackets at the start of each item as indicator of importance.
+          Ignore items starting with "[Internal]", and ignore links to GitHub.
+        PROMPT
+      }.freeze
+
+      def self.run(params)
+        api_token = params[:api_token]
+        prompt = params[:prompt]
+        prompt = PREDEFINED_PROMPTS[prompt] if PREDEFINED_PROMPTS.key?(prompt)
+        question = params[:question]
+
+        headers = {
+          'Content-Type': 'application/json',
+          Authorization: "Bearer #{api_token}"
+        }
+        body = request_body(prompt: prompt, question: question)
+
+        response = Net::HTTP.post(OPENAI_API_ENDPOINT, body, headers)
+
+        case response
+        when Net::HTTPOK
+          json = JSON.parse(response.body)
+          json['choices']&.first&.dig('message', 'content')
+        else
+          UI.user_error!("Error in OpenAI API response: #{response}. #{response.body}")
+        end
+      end
+
+      def self.request_body(prompt:, question:)
+        {
+          model: 'gpt-4o',
+          response_format: { type: 'text' },
+          temperature: 1,
+          max_tokens: 2048,
+          top_p: 1,
+          messages: [
+            format_message(role: 'system', text: prompt),
+            format_message(role: 'user', text: question),
+          ].compact
+        }.to_json
+      end
+
+      def self.format_message(role:, text:)
+        return nil if text.nil? || text.empty?
+
+        {
+          role: role,
+          content: [{ type: 'text', text: text }]
+        }
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Use OpenAI API to generate response to a prompt'
+      end
+
+      def self.authors
+        ['Automattic']
+      end
+
+      def self.return_value
+        'The response text from the prompt as returned by OpenAI API'
+      end
+
+      def self.details
+        <<~DETAILS
+          Uses the OpenAI API to generate response to a prompt.
+          Can be used to e.g. ask it to generate Release Notes based on a bullet point technical changelog or similar.
+        DETAILS
+      end
+
+      def self.examples
+        [
+          <<~EXEMPLE,
+            items = extract_release_notes_for_version(version: app_version, release_notes_file_path: 'RELEASE-NOTES.txt')
+            nice_changelog = openai_generate(
+              prompt: :release_notes, # Uses the pre-crafted prompt for App Store / Play Store release notes
+              question: "Help me write release notes for the following items:\n#{items}",
+              api_token: get_required_env('OPENAI_API_TOKEN')
+            )
```

Comment:
Alternative naming idea:
 - Name the action `openai_prompt(context_prompt:, message:)` or something similar, to highlight that this is focused on text prompts (as opposed to "generating" as in Dall-E and generating images or similar). The thing is, `prompt` here would be used as a verb, while it could be confusing for the name of the action if someone were interpreting it as a noun instead… (is it supposed to return the prompt?!)
 - Name the action `openai_ask(prompt:, question:)` instead of `openai_generate`
 - Use `message:` instead of `question:`…

I think I like `openai_ask(prompt:, question:)` after all, but open to brainstorming and bikeshedding 😅

Reply:
Hmm I'd also vote for `openai_ask(prompt:, question:)`! IMO it sounds more natural and better reflects the "conversation" aspect of the API.

Reply:
Sold! Action renamed in https://github.com/wordpress-mobile/release-toolkit/pull/621/commits/c829b8d6a0b1db0f459c4562ea966da81d287803